test/test_torch.py

```diff
@@ -15149,6 +15149,21 @@ def test_full_out(self, device):
         self.assertEqual(torch.full(o.shape, 1., out=o).dtype, o.dtype)
 
 
+    @onlyOnCPUAndCUDA
+    def test_complex_type_conversions(self, device):
+        dtypes = [torch.float, torch.complex64, torch.complex128]
+        for from_type in dtypes:
+            for to_type in dtypes:
+                from_tensor = torch.randn(4, dtype=from_type, device=device)
+                to_tensor = from_tensor.to(to_type)
+                if from_type.is_complex and not to_type.is_complex:
+                    self.assertEqual(from_tensor.real(), to_tensor, exact_dtype=False)
+                elif not from_type.is_complex and to_type.is_complex:
+                    self.assertEqual(from_tensor, to_tensor.real(), exact_dtype=False)
```

Comment:
to_tensor.real() would return a complex tensor with imaginary part as 0 (which we plan to change to floating point return type in future). but not sure if assertEqual will return true for a fp tensor and a complex tensor

Reply:
Yes, assertEqual is written in such a way as to handle this case correctly.